First_task/First_task/source/Teacher.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace First_task.source
+{
+	class Teacher : Person
+	{
+		private string _degree;
+		private string _subject;
+		private List<string> _students;
+
+		public string Degree
+		{
+			get => _degree;
+			set => _degree = value;
+		}
+
+		public string Subject
+		{
+			get => _subject;
+			set => _subject = value;
+		}
+
+		internal List<string> Students
+		{
+			get => _students;
+			set => _students = value;
+		}
+
+		public Teacher()
+		{
+			Degree = string.Empty;
+			Subject = string.Empty;
+			Students = new List<string>();
+		}
+
+		public Teacher(string name, int age, string degree, string subject, List<string> students) : base(name, age)
+		{
+			Degree = degree;
+			Subject = subject;
+			foreach (string stud in students)
+			{

```

Comment:
Change to add all and add new method assign students that get array of student and made relations betwen student and teacher